agent/otel.go

```diff
@@ -42,6 +42,7 @@ const (
 	packageName    = "github.com/evergreen-ci/evergreen/agent"
 	traceSuffix    = "build/OTelTraces"
 	maxLineSize    = 1024 * 1024
+	batchSize      = 2000
```

Comment:
Is 2000 just a safe batch number? 
Do we want to try to optimize batching for the least amount of uploads? I am not familiar with Honeycomb's pricing model, but if they charge us for uploads it may be worth thinking about.

Reply:
Honeycomb's pricing model revolves around the number of spans we send them. So the batching is just for us, to reduce the overhead of making a request for each span.

I chose 2000 because that seemed like the sweet spot for in a similar issue we faced recently. We were getting errors sending spans from the collectors to refinery because refinery has a similar limit ([DEVPROD-5396](https://jira.mongodb.org/browse/DEVPROD-5396)) and we reduced the batch size until we stopped seeing errors.

Reply:
Come to think of it, the refinery limit is 5MB while this limit is 4MB. I reduced the batch size to 1500 (because it's around 80%)